Philips.CodeAnalysis.Test/Maintainability/Maintainability/AvoidChangingLoopVariablesAnalyzerTest.cs

```diff
@@ -0,0 +1,88 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using Philips.CodeAnalysis.Common;
+using Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability;
+
+namespace Philips.CodeAnalysis.Test.Maintainability.Maintainability
+{
+	/// <summary>
+	/// Test class for <see cref="AvoidChangingLoopVariablesAnalyzer"/>.
+	/// </summary>
+	[TestClass]
+	public class AvoidChangingLoopVariablesAnalyzerTest : DiagnosticVerifier
+	{
+		private const string Correct = @"
+namespace AvoidUsingParametersTest {
+    public class Number {
+        private void MethodA(int a) {
+            for(int i = 0; i < list.Count; i++) {
+                int j = i;
+            }
+        }
+    }
+}";
+
+		private const string CorrectNoLoopVariable = @"
+namespace AvoidUsingParametersTest {
+    public class Number {
+        private void MethodA(int a) {
+            int i = 0;
+            for(; i < list.Count; i++) {
+                int j = i;
+            }
+        }
+    }
+}";
+
+		private const string Wrong = @"
+using System.Collections.Generic;
+namespace AvoidUsingParametersTest {
+    public class Number {
+        private void MethodA(List<int> list) {
+            for(int i = 0; i < list.Count; i++) {
+                i = 4;
+            }
+        }
+    }
+}";
+
+		/// <summary>
+		/// No diagnostics expected to show up
+		/// </summary>
+		[TestMethod]
+		[DataRow("", DisplayName = "Empty"),
+		 DataRow(Correct, DisplayName = nameof(Correct)),
+		 DataRow(CorrectNoLoopVariable, DisplayName = nameof(CorrectNoLoopVariable))]
+		public void WhenTestCodeIsValidNoDiagnosticIsTriggered(string testCode)
+		{
+			VerifyDiagnostic(testCode);
```

Comment:
I recently added "VerifySuccessfulCompilation" to the API, to make it jump out that no diagnostics are expected. No need to change here since there's so much pre-existing code, but consider for future analyzers.

Reply:
Changed